### ISSUE TITLE
error cb() never called!

### ISSUE THREAD

**Issue by xandar6** [npm/npm#19778]
#### I'm opening this issue because:

I get this error when I run command "create-react-app appname" in the console. I have been trying to install react through npm. 
I have never installed node.js before in my pc. following are the steps I took until the error occur.
1) I installed node.js (node-v9.5.0-x64.msi)
2) I ran this command "npm install -g create-react-app"
3) cd to "D:\Dev\web\react" and ran "create-react-app appname" to create a sample project
4) Got this error "error cb() never called!"
[2018-02-08T13_55_39_305Z-debug.log](https://github.com/npm/npm/files/1707397/2018-02-08T13_55_39_305Z-debug.log)

#### How can the CLI team reproduce the problem?

this is the error log
https://gist.githubusercontent.com/kulith6/f88000f07610b4565a801b7ba34ac7df/raw/c489ed5f9c9ff069a75fcd00c2b4aa10decd8314/2018-02-08T13_55_39_305Z-debug.txt

### supporting information:

 - `npm -v` prints:
    5.6.0
 - `node -v` prints:
    v9.5.0
 
 - Windows, OS X/macOS, or Linux?:
   windows 7

**Comment by ivanempire** [comment]
@kulith6 Just to make sure (reading the errors in the log), do you have enough space on the D drive?

**Comment by LucasMembrane** [comment]
I am getting the same message with an instll on linux (fedora 26):

[11:50 al ~]$ npm install -g npm@latest
/home/linuxbrew/.linuxbrew/bin/npm -> /home/linuxbrew/.linuxbrew/lib/node_modules/npm/bin/npm-cli.js
/home/linuxbrew/.linuxbrew/bin/npx -> /home/linuxbrew/.linuxbrew/lib/node_modules/npm/bin/npx-cli.js
+ npm@5.6.0
updated 1 package in 5.329s

[11:51 al ~]$ npm install -g https://github.com/reasonml/reason-cli/archive/3.0.4-bin-linux.tar.gz
npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/al/.npm/_logs/2018-02-08T19_53_03_454Z-debug.log
[11:53 al ~]$ cat /home/al/.npm/_logs/2018-02-08T19_53_03_454Z-debug.log | more
0 info it worked if it ends with ok
1 verbose cli [ '/home/linuxbrew/.linuxbrew/Cellar/node/9.5.0/bin/node',
1 verbose cli   '/home/linuxbrew/.linuxbrew/bin/npm',
1 verbose cli   'install',
1 verbose cli   '-g',
1 verbose cli   'https://github.com/reasonml/reason-cli/archive/3.0.4-bin-linux.
tar.gz' ]
2 info using npm@5.6.0
3 info using node@v9.5.0
4 verbose npm-session c5884cd6f0e00782
5 silly install loadCurrentTree
6 silly install readGlobalPackageData
7 http fetch GET 200 https://github.com/reasonml/reason-cli/archive/3.0.4-bin-li
nux.tar.gz 86046ms
8 error cb() never called!
9 error This is an error with npm itself. Please report this error at:
10 error <https://github.com/npm/npm/issues>
[11:55 al ~]$

**Comment by ptitgege** [comment]
I had same problem on Windows 7 when I run "npm install -g @ angular/cli"

I've tried with both version 8.6.4 and 9.5.0 of NodeJs with same issue.

I solve the problem with version 6.10.2 of NodeJs

**Comment by xandar6** [comment]
@ivanempire That was the issue I had 73mb. It worked after I clear out my drive. Thank you

**Comment by LucasMembrane** [comment]
Why is this closed?
The message that says "npm ERR! This is an error with npm itself. Please report this error at:
npm ERR! https://github.com/npm/npm/issues" would be wrong if this is not an issue because the message says that it is an issue.  But if the message is wrong, then it still is an issue to poor fools like me who cannot solve the problem given the misinformation.  I have tried going back to  6.12 of NodeJs and who do not have  any shortage of disk space.  What to try?

Thanks.

**Comment by ivanempire** [comment]
@LucasMembrane Do you have a log we can look at?

**Comment by LucasMembrane** [comment]
As above:

at /home/al/.npm/_logs/2018-02-08T19_53_03_454Z-debug.log | more
0 info it worked if it ends with ok
1 verbose cli [ '/home/linuxbrew/.linuxbrew/Cellar/node/9.5.0/bin/node',
1 verbose cli '/home/linuxbrew/.linuxbrew/bin/npm',
1 verbose cli 'install',
1 verbose cli '-g',
1 verbose cli 'https://github.com/reasonml/reason-cli/archive/3.0.4-bin-linux.
tar.gz' ]
2 info using npm@5.6.0
3 info using node@v9.5.0
4 verbose npm-session c5884cd6f0e00782
5 silly install loadCurrentTree
6 silly install readGlobalPackageData
7 http fetch GET 200 https://github.com/reasonml/reason-cli/archive/3.0.4-bin-li
nux.tar.gz 86046ms
8 error cb() never called!
9 error This is an error with npm itself. Please report this error at:
10 error https://github.com/npm/npm/issues
[11:55 al ~]$

**Comment by ivanempire** [comment]
@LucasMembrane Oh crap sorry, I'm blind! Can you attach the log file? I don't see any hints of what could be wrong in the snippet....

**Comment by xandar6** [comment]
@LucasMembrane sorry for closing the issue I didn't know how things happen with issues since this is my 1st one

**Comment by sriram-srinivasan** [comment]
I have an identical problem on macOS X Sierra (10.12.6), with the latest npm/node.

0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node',
1 verbose cli   '/usr/local/bin/npm',
1 verbose cli   'install',
1 verbose cli   '-g',
1 verbose cli   'https://github.com/reasonml/reason-cli/archive/3.0.4-bin-darwin.tar.gz' ]
2 info using npm@5.6.0
3 info using node@v9.6.1
4 verbose npm-session 23c04da56d52d3d8
5 silly install loadCurrentTree
6 silly install readGlobalPackageData
7 http fetch GET 200 https://github.com/reasonml/reason-cli/archive/3.0.4-bin-darwin.tar.gz 137084ms
8 error cb() never called!
9 error This is an error with npm itself. Please report this error at:
10 error <https://github.com/npm/npm/issues>

EDIT:

I completely removed all of node/npm/node_modules (all global and local versions) and reinstalled them to the latest (node:9.6.1, npm:5.6.0).  Additionally, I configured the global node configuration to be in my home directory to avoid any sudo/permission issues. I still get the same error. No shortage of disk space either.

**Comment by wakaryry** [comment]
I got the same error.

```
npm --registry=https://registry.npm.taobao.org install -g realm-object-server
npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
```

- It's MacOS high sierra 10.13.3, and it's a new-installed system.
- node 8.9.4

### The log file:
[2018-03-01T06_47_47_548Z-debug.log](https://github.com/npm/npm/files/1770158/2018-03-01T06_47_47_548Z-debug.log)

Any help?

Thanks.

**Comment by ivanempire** [comment]
@wakaryry Do you use brew or any package manager?

**Comment by lengxuehx** [comment]
I encountered the same problem on my Mac and I have tried all methods I can find: upgrade to latest Node, clean cache, remove _node_mudules_ directory, but all have no effect. Eventually, I believed it was the problem of Node environment, so I degraded my Node to an old LTS version 6.14.1, then the problem disappeared. This is what I do:

1.  Install NVM (To get a brand new isolated node environment,  you'd better use NVM to manage your multiple Node environments,  go to [here](https://github.com/creationix/nvm#install-script)  for details)
`curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.8/install.sh | bash`

2. Install Node.js 6.14.1
`nvm install 6.14.1`

3. Switch to the specific Node environment you install above
`nvm use 6.14.1`

4. Install create-react-app
`npm install -g create-react-app`

5. Create your react app
`create-react-app appname`
